### PR TITLE
adding views

### PR DESCRIPTION
**Pull request by romchik2612**

### CODE REVIEW

File: backend/wine_map/views.py [hunk 1]
```diff
@@ -86,3 +87,49 @@ class CommentUpdateView(generics.UpdateAPIView):
...
+        wines = Wine.objects.filter(in_favourites_of__id=user_id)
+        serializer = WineSerializer(wines, many=True)
+
+        return Response(serializer.data)
+
+
+def retrieve_user_id(request: Request) -> int:
+    token = request.headers["Authorization"]
+    token_str = token.split("Bearer ")[1]
+    return AccessToken(token_str)["user_id"]
```
Comment (Nattalli) [inline backend/wine_map/views.py:135]: oh, you find user id in this way. But you can just use `request.user`

File: backend/wine_map/views.py [hunk 2]
```diff
@@ -86,3 +87,49 @@ class CommentUpdateView(generics.UpdateAPIView):
...
+
+
+class FavouriteWines(generics.RetrieveAPIView):
+
+    def get(self, request: Request, *args: tuple, **kwargs: dict) -> Response:
+        user_id = retrieve_user_id(request)
+        wines = Wine.objects.filter(in_favourites_of__id=user_id)
+        serializer = WineSerializer(wines, many=True)
+
+        return Response(serializer.data)
```
Comment (Nattalli) [inline backend/wine_map/views.py:129]: add permission classes here and everywhere

File: backend/wine_map/views.py [hunk 3]
```diff
@@ -90,6 +90,43 @@ class CommentUpdateView(generics.UpdateAPIView):
...
+class FavouriteWinesUpdateView(generics.UpdateAPIView):
+    permission_classes = [AllowAny]
+
+    def update(self, request: Request, *args: tuple, **kwargs: dict) -> Response:
+        wine = Wine.objects.get(pk=self.kwargs["wine_id"])
+        user_id = request.user.id
+        if user_id in wine.in_favourites_of.values_list("id", flat=True):
+            wine.in_favourites_of.remove(user_id)
+        else:
+            wine.in_favourites_of.add(user_id)
```
Comment (antos07) [inline backend/wine_map/views.py:102]:
```suggestion
        user = request.user
        if user.favourite_wines.contains(wine):
            user.favourite_wines.remove(wine)
        else:
            user.favourite_wines.add(wine)
```

File: backend/wine_map/views.py [hunk 4]
```diff
@@ -90,6 +90,43 @@ class CommentUpdateView(generics.UpdateAPIView):
     permission_classes = [IsAuthenticated, IsCommentAuthor]
 
 
+class FavouriteWinesUpdateView(generics.UpdateAPIView):
+    permission_classes = [AllowAny]
+
+    def update(self, request: Request, *args: tuple, **kwargs: dict) -> Response:
+        wine = Wine.objects.get(pk=self.kwargs["wine_id"])
```
Comment (antos07) [inline backend/wine_map/views.py:97]: Add 404 response, when wine is not found

**Review by antos07 — CHANGES_REQUESTED** [review]
I think it could be better to organize endpoints like this:
- `GET /wine/favourites/` to list favourites
- `PUT /wine/favourites/` to put wine into the favourites
- `DELETE /wine/favourites/` to clear favourites
- `DELETE /wine/favourites/<wine_id>` to remove wine from favourites

**Comment by romchik2612** [comment]
> I think it could be better to organize endpoints like this:
> 
> * `GET /wine/favourites/` to list favourites
> * `PUT /wine/favourites/` to put wine into the favourites
> * `DELETE /wine/favourites/` to clear favourites
> * `DELETE /wine/favourites/<wine_id>` to remove wine from favourites

don`t think so, it`s wine entity update - not separate endpoints, and one endpoint for add/remove wine is more comfortable for the front-end

File: backend/wine_map/views.py [hunk 5]
```diff
@@ -90,6 +90,43 @@ class CommentUpdateView(generics.UpdateAPIView):
...
+        return Response(serializer.data)
+
+
+class FavouriteWinesClearView(generics.UpdateAPIView):
+    permission_classes = [AllowAny]
+
+    def update(self, request: Request, *args: tuple, **kwargs: dict) -> Response:
+        user_id = request.user.id
+        for wine in Wine.objects.filter(in_favourites_of__id=user_id):
+            wine.in_favourites_of.remove(user_id)
```
[the remark this answers is not in the document]
Comment (romchik2612) [reply backend/wine_map/views.py:114]: we don`t have user favorite wines, only wine in favourite of

File: backend/wine_map/views.py (hunk 3 above)
Comment (romchik2612) [reply backend/wine_map/views.py:102]: we don`t have user favorite wines

Comment (antos07) [reply backend/wine_map/views.py:102]: We do have))

Django creates a field in a related model. Check out the Wine model defenition - you'll see `in_favourites_of = models.ManyToManyField(get_user_model(), related_name="favourite_wines")`, which creates `favourite_wines` field in the User model

**Review by Nattalli — APPROVED** [review]

**Review by antos07 — APPROVED** [review]

**Pull request merged**